File: frontend/src/utils/product.ts
export interface Product {
  id?: string;
  nombre: string;
  descripcion: string;
  precio: number;
  estado: boolean;
}

export interface ProductDetail extends Product {
  usuarioCreacion?: string | null;
  fechaCreacion?: string | null;
  usuarioModificacion?: string | null;
  fechaModificacion?: string | null;
}

export function normalizeProduct(raw: Record<string, unknown>): Product {
  return {
    id: String(raw.id ?? ""),
    nombre: String(raw.nombre ?? raw.Nombre ?? ""),
    descripcion: String(
      raw.description ?? raw.descripcion ?? raw.Description ?? ""
    ),
    precio: Number(raw.precio ?? raw.Precio ?? 0),
    estado: Boolean(raw.estado ?? raw.Estado),
  };
}

export function normalizeProductDetail(raw: Record<string, unknown>): ProductDetail {
  const base = normalizeProduct(raw);
  return {
    ...base,
    usuarioCreacion: (raw.usuarioCreacion ?? null) as string | null,
    fechaCreacion: (raw.fechaCreacion ?? null) as string | null,
    usuarioModificacion: (raw.usuarioModificacion ?? null) as string | null,
    fechaModificacion: (raw.fechaModificacion ?? null) as string | null,
  };
}

File: Prueba_Tecnica_RedLab.Domain/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Prueba_Tecnica_RedLab.Domain.Entities
{
    public class Product
    {
        public Guid Id {  get; set; }
        public string Nombre { get; set; } = string.Empty;
        public string? Description { get; set; }
        public decimal Precio { get; set; }
        public bool Estado { get; set; }
        public string? UsuarioCreacion { get; set; }
        public DateTime FechaCreacion { get; set; }
        public string? UsuarioModificacion { get; set; }
        public DateTime? FechaModificacion { get; set; }

    }
}
